File: python/armstrong-numbers/armstrong_numbers.py
from math import log10

def is_armstrong_number(number):
    if (number < 0): raise Exception('Only natural number arguments are allowed.')
    if number == 0: return True
    num_digits = int(log10(number)) + 1
    return number == sum([(number // 10**k % 10)**num_digits for k in range(num_digits)])

File: dart/armstrong-numbers/lib/armstrong_numbers.dart
import 'dart:math';

double log10(int n) => log(n) / log(10);

class ArmstrongNumbers {
  bool isArmstrongNumber(int number) {
    if (number < 0) {
      throw ArgumentError('Only natural number arguments are allowed.');
    }
    if (number == 0) {
      return true;
    }
    final num_digits = log10(number).floor() + 1;
    final digits = Iterable<int>.generate(num_digits)
        .map((k) => number ~/ pow(10, k) % 10)
        .toList();
    final sum =
        digits.fold<int>(0, (acc, curr) => acc + pow(curr, num_digits).floor());
    return sum == number;
  }
}
